pipelines/tekton/aiedge-e2e/tasks/retrieve-build-image-info.task.yaml

```diff
@@ -33,18 +33,18 @@ spec:
       echo ;
       echo -n "$(params.model-version)" | tee $(results.model-version.path) ;
       echo ;
-      export DOCKER_IMAGE_REF=$(oc get -n $(params.namespace) -o jsonpath='{.image.dockerImageReference}' imagestreamtag/$(params.model-name):$(params.model-version)) ;
+      export DOCKER_IMAGE_REF=$(skopeo inspect --format '{{.Name}}@{{.Digest}}' docker://image-registry.openshift-image-registry.svc:5000/$(params.namespace)/$(params.model-name):$(params.model-version)) ;
       if [[ $DOCKER_IMAGE_REF != *"$(params.buildah-sha)"* ]]; then
-        echo "ImageStreamTag doesn't contain the correct image SHA"
+        echo "Candidate image tag does not contain the correct manifest digest after push"
         exit 1 ;
       fi
       echo -n $DOCKER_IMAGE_REF | tee $(results.internal-image-url.path) ;
       echo ;
-      oc get -n $(params.namespace) -o jsonpath='{.image.dockerImageMetadata.Size}' imagestreamtag/$(params.model-name):$(params.model-version) | tee $(results.internal-image-size.path) ;
+      echo $(($(skopeo inspect --format '{{range .LayersData}}+{{.Size}}{{end}}' docker://$DOCKER_IMAGE_REF))) | tee $(results.internal-image-size.path) ;
```

Comment:
Just FYI, when I tried this, when inspecting `docker://` image, it returns uncompressed size, when using `docker-daemon:sha256:<sha256-of-local-docker-registry>` it seems to return the full size.
E.g.:
`skopeo inspect docker://docker.io/library/fedora@sha256:830a6bcb5c9b30bfa77479b747a904bed02151e9de7ceeb8a1d117b574571fa4` (fedora:latest)
returns 
``` json
"Size": 64614010
```
whereas
`skopeo inspect docker-daemon:sha256:5f878605acd6f24f3f11e4085544caf61b046378b31d6b3aa0677a7a3dbe0a17`
which is my local IMAGE ID for the same image returns
```json
"Size": 182162432
```
and docker inspect for that same image returns
```json
"Size": 176658692
```
The last 2 outputs are from local image, the first one is for a remote, compressed image.

I think for now it's OK. Just in case we need to have constraints about image size we should be more careful what we are measuring.

Reply:
Yeah, that's interesting. I was actually initially just going to omit this bit of information (and maybe one or two others also), since I'm not sure why we want it or how we'll use it.

It does appear to be the compressed size that we were getting previously: I imported an image into an imagestream & compared. E.g.:

```
# import image to imagestream
oc import-image fedora --from=quay.io/fedora/fedora --all --confirm

# then check the size of a tag in the imagestream
oc get -o jsonpath='{.image.dockerImageMetadata.Size}' imagestreamtag/fedora:39
68672016

# compare that to the new approach of adding layer sizes in original repo
echo $(($(skopeo inspect --format '{{range .LayersData}}+{{.Size}}{{end}}' docker://quay.io/fedora/fedora:39)))
68670700
```

^there's a slight discrepancy (~1KiB), which I'd guess is the manifest referring to the layers?

Reply:
This metric is for the future when we are planning to check the real image size and have a test for it, if the image is above the limit, the pipeline would likely fail or something similar.

Also for my other images with more than one layer the numbers are slightly off, roughly by 5 MB. Interestingly skopeo reports bigger sum than docker inspect. But it may be also due to rounding or different way of calculation by skopeo.